p2p/host/basic/mockgen_private.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+DEST=$2
```

Comment:
Did you copy this from quic-go? I'm curious how deeply this needs to be reviewed

Reply:
I did, and it's super ugly. I figured out a better solution though (thanks ChatGPT): We can just export the type in a file that has a `gomock` build flag, and then run `mockgen` with the `-build_flags="-tags=gomock"` command line parameter.
That way, `gomock` will think that it's an exported type, whereas the final build (and, very importantly, your IDE) will not.

Reply:
In fact, I should do the same in quic-go. This bash script has been annoying me for a long time!